src/hotspot/share/gc/shenandoah/shenandoahFreeSet.hpp

```diff
@@ -341,6 +341,7 @@ class ShenandoahFreeSet : public CHeapObj<mtGC> {
 
   void recycle_trash();
   void log_status();
+  void log_status_under_lock();
```

Comment:
Add a brief 1-line documentation comment for each of the methods, mentioning the preconditions & what they do.

Reply:
I forgot. Shouldn't `log_status` remove the enablement check, and assert that logging is enabled?

Reply:
Discussed it in another conversation.